Modify rule S5485: Expand and adjust for LaYC

pdschbrt opened the pull request:
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

tomasz-kaminski-sonarsource commented:
I think that the current rule title "Appropriate arguments should be passed to stream functions" (in metadata file), does not meet LaYC standards.

tomasz-kaminski-sonarsource commented on rules/S5485/cfamily/rule.adoc:
```diff
@@ -1,47 +1,149 @@
-* ``++FILE*++`` should be checked for null before being used in a function that accesses the file content
-* The third argument of ``++fseek++`` must be ``++SEEK_SET++``, ``++SEEK_END++``, or ``++SEEK_CUR++``
+The standard C library includes a number of functions for handling I/O streams.
+These functions require appropriate parameters.
```
I think that it would be nicer, to formulate it, as this functions put restrictions on the values of the parameters, and have bulled list of the restrictions.
If they are not all, for example `ftell` with not opened file is probably also bad, I would say including, but not limited to.

tomasz-kaminski-sonarsource commented on rules/S5485/cfamily/rule.adoc:
```diff
@@ -1,47 +1,149 @@
-ftell(file2); // Noncompliant, file could be NULL
-if (file2) {
-  fseek(file2, 1, 3); // Noncompliant, third argument should either be SEEK_SET, SEEK_CUR or SEEK_END
+Besides the aforementioned availability issues of the application, any attempts to dereference a ``++NULL++`` pointer made by the C library functions can, in rare circumstances, also results in security issues.
```
I would sync this null pointer dereference with the other rule, for consistency.

pdschbrt commented:
Hey @tomasz-kaminski-sonarsource, I've addressed your comments and harmonized this rule with the very similar rule that we worked on. Let me know what you think.

tomasz-kaminski-sonarsource commented on rules/S5485/cfamily/metadata.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "title": "Appropriate arguments should be passed to stream functions",
+  "title": "Sufficiently restricted arguments should be passed to stream functions",
```
I think I like appropariate here more, after rereading. I know this was my suggestion, but this sound akward.

tomasz-kaminski-sonarsource replied:
Could you try to consult with best practies team on this tree "Appropariate" rule titles? They feel as not matching guidelines for me, but I think that very subjective.
If you change this one, I would suggest adjusting previous rule in this PR>

tomasz-kaminski-sonarsource replied:
There is style guide on titles here: https://docs.sonarsource.com/sonarqube/latest/extension-guide/adding-coding-rules/#coding-rule-guidelines

pdschbrt replied:
Sure. We, in fact, have a whole bunch of rules that are concerned with `appropriate` parameters. I'll double-check with best practices 👍🏻

tomasz-kaminski-sonarsource replied:
I like the only valid approach. Much better that appropariate.

tomasz-kaminski-sonarsource commented on rules/S5485/cfamily/rule.adoc:
```diff
@@ -1,47 +1,150 @@
+Passing insufficiently restricted arguments to standard C library functions for handling I/O streams results in undefined behavior.
```
Same here.

tomasz-kaminski-sonarsource commented on rules/S5485/cfamily/rule.adoc:
```diff
@@ -1,47 +1,150 @@
 
-The standard C library includes a number of functions for handling streams. If not called correctly, these functions can have undefined behavior. More specifically:
+The standard C library includes a number of functions for handling I/O streams.
+These functions put certain restrictions on the values of their parameters, which are listed in the following:
```
I think there are more restriction placed. I believe FILE must be actually correct file handle. So I would say something that this is subset of restrictions.

pdschbrt replied:
True. Let me try to rephrase this slightly.

tomasz-kaminski-sonarsource replied:
Looks good.

tomasz-kaminski-sonarsource commented on rules/S5485/cfamily/rule.adoc:
```diff
@@ -1,47 +1,150 @@
+#include <stdio.h>
+
+size_t get_file_size() {
+  FILE *f = fopen("example_file.txt", "r");
```
I would just add comment here, saying:
```
// may be null if fopen operation failed
```
Then I would just remove the description of code from above.

tomasz-kaminski-sonarsource reviewed (changes requested):
Looks good. Just few small suggestions.

pdschbrt commented:
How about this?

tomasz-kaminski-sonarsource reviewed (approved):
Looks good to me now.

pdschbrt commented:
Excellent. Thanks for your thorough review.
